frontend/src/Components/TodoList.jsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import { useState } from 'react';
+import { Box, Grid, Typography } from '@mui/material';
+import TodoCard from './TodoCard';
+import CreateItemForm from './CreateItemForm';
+
+const columnStyles = {
+  sx: {
+    borderRadius: 3,
+    backgroundColor: 'rgb(246, 247, 248)',
+    paddingBottom: '1.5%',
+    marginBottom: '1%',
+    boxShadow: 1,
+  },
+};
+
+const columnContainerStyles = {
+  style: {
+    maxHeight: '100%',
+    maxWidth: '100%',
+    width: '100%',
+    paddingTop: '2%',
+    paddingBottom: '2%',
+
+  },
+};
+
+const formContainerStyles = {
+  style: {
+    width: '60%', marginBottom: '.5rem',
+  },
+};
+
+const columnNameStyles = {
+  style: {
+    paddingLeft: '1.6rem',
+  },
+};
+
+const todoCardItemStyle = {
+  style: { width: '100%' },
+};
+
+export default function TodoList() {
+  const [todoItems, setTodoItems] = useState({});
+  const [allTimeTodoCount, setAllTimeTodoCount] = useState(0);
+
+  // eslint-disable-next-line max-len
+  const inProgressTodos = () => Object.values(todoItems).filter((item) => item.isCompleted === false);
+  const completedTodos = () => Object.values(todoItems).filter((item) => item.isCompleted === true);
+  const createTodoItem = (args) => {
+    const { title, description } = args;
+    const newTodo = {
+      title,
+      description,
+      isCompleted: false,
+      id: allTimeTodoCount.toString(),
+      index: Object.values(todoItems).length,
+    };
+    setTodoItems((prevTodoItems) => ({ ...prevTodoItems, [newTodo.id]: newTodo }));
+    setAllTimeTodoCount(allTimeTodoCount + 1);
+  };
+  const markTodoItem = (id) => {
+    setTodoItems((prevTodoItems) => ({
+      ...prevTodoItems,
+      [id]: {
+        ...prevTodoItems[id],
+        isCompleted: !prevTodoItems[id].isCompleted,
+      },
+    }));
+  };
+
+  const deleteTodoItem = (id) => {
+    const newToDoItems = { ...todoItems };
+    delete newToDoItems[id];
+    setTodoItems(newToDoItems);
+  };
+
+  const sortItems = (items) => items.sort((a, b) => a.index - b.index);
+
+  const updateTodoItem = (item) => {
+    setTodoItems((prevItems) => ({ ...prevItems, [item.id]: { ...item } }));
+  };
+
+  React.useEffect(() => {
```

Comment:
It would be better to import just the pieces you need rather than all of the react library. This is true of most libraries, and the reason is that some builders have a mechanism called "tree shaking" where only the pieces used will end up in the resulting code bundle. And by pulling in the entirety of the library it makes the build think you are using everything when it does the tree shaking.

Reply:
Makes sense, thanks 😄 ! I've fixed this now by replacing the import statement with `import { useState, useEffect } from 'react';` and then called `useEffect()` instead of `React.useEffect()`. 

I removed the unnecessary React imports in other files as well.